src/Uno.UI/UI/Xaml/Media/ImageSource.cs

```diff
@@ -247,9 +248,20 @@ private set
 
 		private protected async Task<Stream> OpenStreamFromUriAsync(Uri uri, CancellationToken ct)
 		{
-			_httpClient ??= new HttpClient();
-			var response = await _httpClient.GetAsync(uri, HttpCompletionOption.ResponseContentRead, ct);
-			return await response.Content.ReadAsStreamAsync();
+			if (uri.IsFile)
+			{
+#pragma warning disable SYSLIB0014 // Type or member is obsolete - It suggests to use HttpClient, which doesn't work for file Uri scheme.
+				var request = WebRequest.Create(uri);
+#pragma warning restore SYSLIB0014 // Type or member is obsolete
+				var response = await request.GetResponseAsync();
+				return response.GetResponseStream();
```

Comment:
We probably should be using `File` APIs instead. Uri has `LocalPath` that we can use.

Reply:
@Youssef1313

Reply:
Done